### ISSUE TITLE
7.11 APIs who don’t need a body now return an error when provided with one

### ISSUE THREAD

**Issue by m-a-leclercq** [elastic/elasticsearch#69532]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.11.1 -  docker image

**Description of the problem including expected versus actual behavior**:
Hi,

I have been upgrading one of my clusters to 7.11.1 and much to my surprise there is a behavior change that I haven't been able to find in the breaking changes or in the "what's new" page for this release.
```
GET _cat/health
{
}
```
Now returns
```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "request [GET /_cat/health] does not support having a body"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "request [GET /_cat/health] does not support having a body"
  },
  "status" : 400
}
```

**Expected behavior:**
Elastic should return the actual cluster state.

Now, the problem was in my ansible playbook because my `uri` module had the `body_format: json` property set when checking for the cluster's health so it was a good opportunity to clean up my code.

Since I haven't found any mentions of this behavior change in the documentation or in the github issues, I thought I would put it out there for everyone as a head's up. Maybe someone in the elastic team can add it to the documentation?

This also affects `POST _flush/synced` and many other APIs I suppose.

This is confirmed to work in 7.7.0 and I believe 7.10.2 too but it will return an error in 7.11.1

Original post : https://discuss.elastic.co/t/7-11-apis-who-dont-need-a-body-now-return-an-error-when-provided-with-one/265324

**Comment by jaymode** [comment]
Thanks for opening the issue @m-a-leclercq. This was indeed a deliberate change as noted in https://github.com/elastic/elasticsearch/pull/66043. These requests should have been deprecated in 6.x by https://github.com/elastic/elasticsearch/pull/37534. https://github.com/elastic/elasticsearch/pull/37504 was the change for 7.0+ that IIRC was later broken.

I do believe that we should document this as it may be surprising.

**Comment by jrodewig** [comment]
Closed by #69565.
